components/pd_client/src/util.rs

```diff
@@ -239,11 +246,17 @@ impl Client {
                         self.enable_forwarding,
                     )
                     .await
-            };
-            slow_log!(start.elapsed(), "try reconnect pd");
-            (fut, start)
+            }
         };
 
+        {
+            slow_log!(start.elapsed(), "try reconnect pd");
+            let mut inner = self.inner.wl();
+            if start > inner.last_try_reconnect {
```

Comment:
This is always true.

Reply:
`start` is not taken in the write lock, I don't think it can be guaranteed.